api/server/router/image/image_routes.go

```diff
@@ -226,16 +226,21 @@ func (s *imageRouter) getImagesJSON(ctx context.Context, w http.ResponseWriter,
 	if err != nil {
 		return err
 	}
-
-	version := httputils.VersionFromContext(ctx)
-	if versions.LessThan(version, "1.41") {
+	var sharedSize bool
+	if versions.LessThan(httputils.VersionFromContext(ctx), "1.41") {
 		filterParam := r.Form.Get("filter")
 		if filterParam != "" {
 			imageFilters.Add("reference", filterParam)
 		}
+	} else {
+		sharedSize = httputils.BoolValue(r, "shared-size")
 	}
 
-	images, err := s.backend.Images(imageFilters, httputils.BoolValue(r, "all"), false)
+	images, err := s.backend.Images(ctx, types.ImageListOptions{
```

Comment:
for the first "refactor" commit message that adds the `ImageListOptions` here, I think "support more control" is a bit confusing, because in that commit it's not yet adding "more control". Perhaps something like:

```
API, daemon/images: add ImageListOptions and pass context

This makes it easier to add more options to the backend without having to change
the signature.

While we're changing the signature, also adding a context.Context, which is not
currently used, but probably should be at some point.
```